src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs

```diff
@@ -730,6 +733,75 @@ public async Task PostImportAsync()
             }
         }
 
+        private void IgnoreDatabaseItemTypesIfCurrentVersionIsOlderThenClientVersion(DocumentsOperationContext context, ref BlittableJsonReaderObject blittableJson)
+        {
+            DynamicJsonValue djv = null;
+
+            if (blittableJson.TryGet(nameof(DatabaseSmugglerOptions.OperateOnTypes), out string operateOnTypes) &&
+                operateOnTypes != null && 
+                Enum.TryParse(typeof(DatabaseItemType), operateOnTypes, out _) == false)
+            {
+                CheckClientVersion();
+
+                var itemsTypes = DatabaseItemType.None;
+                var types = operateOnTypes.Split(", ");
+
+                foreach (var type in types)
+                {
+                    if (Enum.TryParse(typeof(DatabaseItemType), type, true, out var result) == false || result is DatabaseItemType existing == false)
+                        continue;
+
+                    itemsTypes |= existing;
+                }
+
+                djv = new DynamicJsonValue(blittableJson)
+                {
+                    [nameof(DatabaseSmugglerOptions.OperateOnTypes)] = itemsTypes
+                };
+            }
+
+            if (blittableJson.TryGet(nameof(DatabaseSmugglerOptions.OperateOnDatabaseRecordTypes), out string operateOnDatabaseRecordTypes) &&
+                operateOnDatabaseRecordTypes != null && 
+                Enum.TryParse(typeof(DatabaseRecordItemType), operateOnDatabaseRecordTypes, out _) == false)
+            {
+                CheckClientVersion();
+
+                var databaseRecordItemsTypes = DatabaseRecordItemType.None;
+                var types = operateOnDatabaseRecordTypes.Split(", ");
+
+                foreach (var type in types)
+                {
+                    if (Enum.TryParse(typeof(DatabaseRecordItemType), type, true, out var result) == false || result is DatabaseRecordItemType existing == false)
+                        continue;
+
+                    databaseRecordItemsTypes |= existing;
+                }
+
+                if (djv == null)
+                    djv = new DynamicJsonValue(blittableJson);
+
+                djv[nameof(DatabaseSmugglerOptions.OperateOnDatabaseRecordTypes)] = databaseRecordItemsTypes;
+            }
+
+            if (djv == null)
+                return;
+
+            blittableJson.Modifications = djv;
+            using (var old = blittableJson)
+            {
+                blittableJson = context.ReadObject(blittableJson, "item-types");
+            }
+        }
+
+        private void CheckClientVersion()
+        {
+            if (RequestRouter.TryGetClientVersion(HttpContext, out var version) == false)
+                throw new VersionNotFoundException("Could not find client version");
```

Comment:
Same comment as last time. You are hiding behind those exceptions the real one. 

What is the problem that you are trying to solve in this issue:
1. That Client-Version is not send in headers?
2. That value supplied in OperateOnTypes and/or OperateOnDatabaseRecordTypes is not parsable?

Reply:
This comment applies to both of the 'if' statements in this method.

Reply:
done